index.js

```diff
@@ -1 +1,31 @@
-// Do work!
\ No newline at end of file
+/* eslint-disable nonblock-statement-body-position */
+function validatePassword(password) {
+  let allRequirementsMet = false
+  let specCharacters = '!"#$%&/' / '()*+,-.'
+
+  // Determine if the password length is more than or equal to 8 then..
+  if (password.length >= 8) {
+    // Using ASCII index and charCodeAt method (thanks to Jkearns!)
+    // Test further requirements
+    // Test for uppercase characters
+    for (let i = 0; i < password.length; i++) {
+      if (password.charCodeAt(i) >= 65 && password.charCodeAt(i) <= 90)
+      // Test for lowercase characters
+        for (let i = 0; i < password.length; i++) {
+          if (password.charCodeAt(i) >= 97 && password.charCodeAt(i) <= 122)
+          // Test for numeric value
```

Comment:
Given a password of 10 chars, we will loop 10 times x 10 x 10 in this deeply nested inner loop.

We are now looping 1000 times. We are looping the same loop within itself.